pgactivity/data.py

```diff
@@ -145,28 +146,40 @@ def pg_is_local_access(self) -> bool:
         Verify if the user running pg_activity can access
         system information for the postmaster process.
         """
-        try:
-            query = queries.get("get_pid_file")
-            with self.pg_conn.cursor() as cur:
-                # This query doesn't crash when the user doesn't have the
-                # required privilege to acces the data_directory guc
-                # it will just return an empty string
+        logger = logging.getLogger("pgactivity")
+        query = queries.get("get_data_directory")
+        with self.pg_conn.cursor() as cur:
+            try:
                 cur.execute(query)
-                ret = cur.fetchone()
-            pid_file = ret["pid_file"]
-            with open(pid_file, "r") as fd:
-                pid = fd.readlines()[0].strip()
-                try:
-                    proc = psutil.Process(int(pid))
-                    proc.io_counters()
-                    proc.cpu_times()
-                    return True
-                except psutil.AccessDenied:
-                    return False
-                except Exception:
-                    return False
-        except Exception:
+            except InsufficientPrivilege:
+                logger.info(
+                    "Insufficient privilege to show data_directory. System counters are disabled."
+                )
+                return False
+            ret = cur.fetchone()
+        try:
+            pid_file = f"{ret['data_directory']}/postmaster.pid"
+        except PermissionError:
+            logger.info("Access Denied to the pidfile. System counters are disabled.")
             return False
+        with open(pid_file, "r") as fd:
+            pid = fd.readlines()[0].strip()
```

Comment:
You can change this as:
```suggestion
            pid = fd.readline().strip()
```
by the way.